packages/synchro-charts/src/components/sc-dial/sc-dial-base/sc-dial-svg.tsx

```diff
@@ -62,13 +92,91 @@ export class ScDialSvg {
       labelSize: newSize?.labelSize || LABEL_SIZE,
       unitSize: newSize?.unitSize || UNIT_SIZE,
     };
+    this.sizeChanged = true;
+  }
+
+  @Watch('percent')
+  onPercentsChanged() {
+    this.changeRing();
+  }
+
+  changeRing() {
+    const ringD: DefaultArcObject = {
+      innerRadius: DIAMETER,
+      outerRadius: DIAMETER - this.sizeConfig.dialThickness,
+      padAngle: RADIAN / 2,
+      startAngle: 0,
+      endAngle: 0,
+    };
+    const unitRadian = Math.PI * 2;
+    const angle1 = unitRadian * this.percent;
+    const angle2 = unitRadian * (1 - this.percent);
+    const currentAngle = RADIAN;
+    const endAngle1 = currentAngle + angle1;
+    const endAngle2 = endAngle1 + angle2;
+
+    const colorArc = arc()
+      .cornerRadius(CORNER_RADIUS)
+      .startAngle(currentAngle)
+      .endAngle(endAngle1);
+    const defaultArc = arc()
+      .cornerRadius(CORNER_RADIUS)
+      .startAngle(endAngle2)
+      .endAngle(endAngle1);
+
+    transition()
+      .duration(DURATION)
+      .ease(easeCubicInOut)
+      .attrTween('d', () => {
+        const endAngleColor = colorArc.endAngle()(ringD);
+        const startAngleColor = colorArc.startAngle()(ringD);
+
+        const endAngleDefault = defaultArc.endAngle()(ringD);
+        const startAngleDefault = defaultArc.startAngle()(ringD);
+
+        const interpolateColor = interpolate(
+          { startAngle: this.oldColorAngle.startAngle, endAngle: this.oldColorAngle.endAngle },
+          { startAngle: startAngleColor, endAngle: endAngleColor }
+        );
+
+        const interpolateDefault = interpolate(
+          { startAngle: this.oldDefaultAngle.startAngle, endAngle: this.oldDefaultAngle.endAngle },
+          { startAngle: startAngleColor, endAngle: endAngleColor }
+        );
+
+        this.oldColorAngle.startAngle = startAngleColor;
```

Comment:
these values should instead be set local to the method, in scope of the function, as opposed to being stored in state within the component. this also removes the need for resetting them to 'blank' after the function is complete.

Reply:
If these values are out of the state within the component, it will loss of animation  after percent has changed.